src/storage/database.rs

```diff
@@ -91,22 +110,69 @@ mod tests {
                     content: "Hello world!".bytes().collect(),
                     compression: None,
                 },
-                backend.get("dir/foo.txt")?
+                backend.get("dir/foo.txt", std::usize::MAX)?
             );
 
             // Test that other files are not returned
             assert!(backend
-                .get("dir/bar.txt")
+                .get("dir/bar.txt", std::usize::MAX)
                 .unwrap_err()
                 .downcast_ref::<PathNotFoundError>()
                 .is_some());
             assert!(backend
-                .get("foo.txt")
+                .get("foo.txt", std::usize::MAX)
                 .unwrap_err()
                 .downcast_ref::<PathNotFoundError>()
                 .is_some());
 
             Ok(())
         });
     }
+
+    #[test]
+    fn test_get_too_big() {
```

Comment:
I know I've said it a lot, but maybe more buffering tests here?

Reply:
What do you mean?